pkgs/build-support/dotnet/make-nuget-deps/default.nix

```diff
@@ -1,15 +1,21 @@
-{ linkFarmFromDrvs, fetchurl }:
+{ linkFarmFromDrvs, substituteAll, fetchurl, runCommand, zip, dotnet-sdk }:
 { name, nugetDeps ? import sourceFile, sourceFile ? null }:
 linkFarmFromDrvs "${name}-nuget-deps" (nugetDeps {
   fetchNuGet = { pname, version, sha256 ? "", hash ? ""
     , url ? "https://www.nuget.org/api/v2/package/${pname}/${version}" }:
-    fetchurl {
-      name = "${pname}.${version}.nupkg";
+    let from-source = fetchurl {
+      name = "${pname}.${version}.nupkg.${url}";
       # There is no need to verify whether both sha256 and hash are
       # valid here, because nuget-to-nix does not generate a deps.nix
       # containing both.
       inherit url sha256 hash;
     };
+    in
+    runCommand "${pname}.${version}.nupkg" {
+      nativeBuildInputs = [ zip ];
+      ORIGINAL_NUPKG = from-source;
+      # DOTNET_EXE = "${dotnet-sdk}/bin/dotnet";
```

Comment:
Not yet been able to verify signatures, so skipping that for now.

Reply:
we should probably just do that in nuget-to-nix.  no point in doing it again after that.

it might even be done already during restore, but I'm not sure.

Reply:
That seems reasonable. I'll delete that completely from this PR.

Reply:
Do you think revocation lists would be distributed in the sdk, or would it always need to be online in order to check for revocation?

Reply:
This depends on `NUGET_CERT_REVOCATION_MODE`. I generally work with that variable set to `offline` so that it checks only against the SDK's bundled lists, but that's because I'm usually behind a firewall and don't have access to the Internet.